### ISSUE TITLE
Implementation of real_sph_harmonic

### ISSUE THREAD

**Issue by vineetbansal** [ComputationalCryoEM/ASPIRE-Python#3]
Comments in code indicate that the following is possibly a more "textbook" implementation of real_sph_harmonic. Discuss with @janden and either make this the default, or discard altogether:

```
    @staticmethod
    def real_sph_harmonic2(j, m, theta, phi):
        # TODO: This needs to be made the default implementation of real_sph_harmonic
        # Get the real basis of spherical harmonics in terms of their complex analogues
        # Note that sph_harm reverses the notation of theta/phi
        # 0.707106 = 1/sqrt(2)
        y_m_l = sph_harm(m, j, phi, theta)
        if m == 0:
            result = y_m_l
        else:
            y_minus_m_l = sph_harm(-m, j, phi, theta)
            if m < 0:
                result = 0.707106j * (y_m_l - (-1)**m * y_minus_m_l)
            else:
                result = 0.707106 * (y_minus_m_l + (-1)**m * y_m_l)

        return np.real(result)
```

**Comment by janden** [comment]
So if I understand correctly, the difference between this implementation and `real_sph_harmonic` is that this one (a) uses the complex `sph_harm` and converts it to a set of real functions and (b) replaces `np.sqrt(2)` with `0.707106`. I don't see why we should do (b) and while I agree that (a) is clearer than what we have, it's also computationally more intensive, since we compute everything in the complex  domain only to convert it back to real values. Since complex values take twice the memory and generally twice the computational effort, this can make a difference. Perhaps a benchmark would sort this out?

**Comment by junchaoxia** [comment]
It looks that we will encounter problem for the current implementation of `real_sph_harmonic` when j is large such as 64:
https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/develop/src/aspire/basis/basis_utils.py#L116
1) The np.prod could become very large and overflow, as in line
    https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/develop/src/aspire/basis/basis_utils.py#L138
2) The  many values of `lpmv` functions also become `inf` in line, 
https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/develop/src/aspire/basis/basis_utils.py#L129

But these two might be cancelled if we use the `scipy.special.sph_harm` function directly. I am reimplementing it using this function to see whether it is OK for large j values.

**Comment by janden** [comment]
Makes sense to use the library function if it works.

**Comment by junchaoxia** [comment]
The issue was addressed in PR #346. Closing now.
